bindings/matrix-sdk-ffi/src/event.rs

```diff
@@ -200,7 +201,21 @@ impl TryFrom<AnySyncMessageLikeEvent> for MessageLikeEventContent {
                     in_reply_to_event_id,
                 }
             }
-            AnySyncMessageLikeEvent::RoomRedaction(_) => MessageLikeEventContent::RoomRedaction,
+            AnySyncMessageLikeEvent::RoomRedaction(c) => {
+                let (redacted_event_id, reason) = match c {
+                    SyncRoomRedactionEvent::Original(o) => {
+                        if let Some(id) = o.redacts {
+                            (Some(id.to_string()), None)
+                        } else if let Some(id) = o.content.redacts {
+                            (Some(id.to_string()), o.content.reason)
+                        } else {
+                            (None, None)
+                        }
```

Comment:
You can pass the reason as `o.content.reason` all the time, it's optional but it's in the same place for all versions of the room (if I read Ruma's code correctly).